src/map/clif.c

```diff
@@ -20377,6 +20381,19 @@ static void clif_monster_hp_bar(struct mob_data *md, struct map_session_data *sd
 #endif
 }
 
+static bool clif_show_monster_hp_bar(const struct mob_data *md, struct block_list *bl) {
+	
+	if (status_get_hp(bl) < status_get_max_hp(bl)) {
+		if ((battle_config.show_monster_hp_bar & 1 && !(md->class_ == MOBID_EMPELIUM) && !(md->status.mode & MD_BOSS))
+			|| (battle_config.show_monster_hp_bar & 2 && (md->class_ == MOBID_EMPELIUM))
```

Comment:
access battle configs using `battle->bc->show_monster_hp_bar` for new code.

Reply:
Please see changes in last commit